File: remocra/src/main/java/fr/sdis83/remocra/domain/remocra/Metadonnee.java
package fr.sdis83.remocra.domain.remocra;

import javax.persistence.Column;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.validation.constraints.NotNull;

import org.springframework.roo.addon.javabean.RooJavaBean;
import org.springframework.roo.addon.jpa.activerecord.RooJpaActiveRecord;
import org.springframework.roo.addon.json.RooJson;
import org.springframework.roo.addon.tostring.RooToString;

@RooJavaBean
@RooToString
@RooJson
@RooJpaActiveRecord
public class Metadonnee {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column
    @NotNull
    private String titre;

    @Column
    private String resume;

    @Column
    private String urlVignette;

    @Column
    private String urlFiche;

    @ManyToOne
    @NotNull
    private Thematique thematique;

}

File: remocra/src/main/webapp/js/app/remocra/model/Metadonnee.js
Ext.define('Sdis.Remocra.model.Metadonnee', {
    extend: 'Ext.data.Model',

    fields: [ {
        name: 'id',
        type: 'int',
        useNull: true
    }, {
        name: 'titre',
        type: 'string'
    }, {
        name: 'resume',
        type: 'string'
    }, {
        name: 'urlVignette',
        type: 'string'
    }, {
        name: 'urlFiche',
        type: 'string'
    }, {
        name: 'thematique',
        type: 'int',
        mapping: function(data, reader) {
            if (data['thematique']) {
                return data['thematique']['id'];
            }
        },
        persist: false
    }, {
        name: 'codeExport',
        type: 'string'
    } ]
});